packages/ember/tests/routing/router_service_test.js

```diff
@@ -0,0 +1,432 @@
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        },
+
+        afterModel() {
+          this.transitionTo('parent.sister');
+        }
+      }));
+
+      return this.visit('/child').then(() => {
+        assert.equal(routerService.get('currentURL'), '/sister');
+      });
+    }
+
+    ['@test RouterService#currentURL is correctly set on each transition'](assert) {
+      assert.expect(3);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.child', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/child')
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/child');
+
+          return this.visit('/sister');
+        })
+        .then(() => {
+          assert.equal(routerService.get('currentURL'), '/sister');
+
+          return this.visit('/brother');
+        })
+        .then(() => {
+            assert.equal(routerService.get('currentURL'), '/brother');
+        });
+    }
+
+    ['@test RouterService#rootURL is correctly set to the default value'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.index', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/').then(() => {
+        assert.equal(routerService.get('rootURL'), '/');
+      });
+    }
+
+    ['@test RouterService#rootURL is correctly set to a custom value'](assert) {
+      assert.expect(1);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.index', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          set(this.router, 'rootURL', '/test');
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/').then(() => {
+        assert.equal(routerService.get('rootURL'), '/test');
+      });
+    }
+
+    ['@test RouterService#location is correctly delegated from router:main'](assert) {
+      assert.expect(2);
+
+      let routerService;
+
+      this.buildRouterMap();
+
+      this.registerRoute('parent.index', Route.extend({
+        routerService: inject.service('router'),
+        init() {
+          this._super();
+          routerService = get(this, 'routerService');
+        }
+      }));
+
+      return this.visit('/').then(() => {
+        let location = routerService.get('location');
+        assert.ok(location);
+        assert.ok(location instanceof Ember.NoneLocation);
```

Comment:
This should be imported from `ember-routing/locations/none` (using global `Ember` within Ember or the test suite isn't allowed). This is currently causing the linting tests to fail...